[UX] Quick filters docs (#208)
mautic/developer-documentation-new: docs/design/quick_filters.rst

Reviewer: Please see notes on other PRs about the heading nesting:

```
H1 ###
H2 ***
H3 ===
H4 ---
H5 ~~~
```

Quick filters for searches
==========================

Quick filters provide an efficient way to search using existing search commands. This documentation outlines how to create and implement new quick filters for Mautic searches.

Implementation overview
-----------------------

> Quick filters are implemented using a combination of JavaScript and Twig templates. The process involves three main components:

Reviewer: ```suggestion
Mautic implements quick filters using a combination of JavaScript and Twig templates. The process involves three main components:
```

1. JavaScript function for applying the filter
2. Twig template for rendering filter buttons
3. Array of PHP code for defining filter options

> JavaScript functionality

Reviewer: ```suggestion
JavaScript features available
```

Author: it could be strange to talk about features (things that generate a possible action on the UI) when in fact we're talking about the gears from the system

what do you think?

------------------------

The ``Mautic.listQuickFilter`` function is responsible for applying the quick filter:

.. code-block:: javascript

   Mautic.listQuickFilter = function (element) {
       const filterValue = element.dataset.filter;
       const searchInput = document.getElementById('list-search');
       searchInput.value = filterValue;
       const enterKeyEvent = new KeyboardEvent('keyup', {
           keyCode: 13
       });
       searchInput.dispatchEvent(enterKeyEvent);
   }

This function performs the following actions:

1. Retrieves the filter value from the clicked element's ``data-filter`` attribute
2. Sets the search input field's value to the filter value
3. Simulates an Enter key press to trigger the search

Twig template
-------------

> The quick filter buttons are rendered using a Twig template:

Reviewer: ```suggestion
Mautic renders the quick filter buttons using a Twig template:
```

              data-toggle="tooltip"
              data-placement="top"
              data-original-title="{{ quickFilter.tooltip|trans }}">
               <i class="{{ quickFilter.icon }}"></i>
               {{ quickFilter.label|trans }}
           </a>
       {% endfor %}
   </div>
   {% endif %}

This template iterates through the provided quick filters and creates clickable labels for each one on the toolbar.

Implementing quick filters
--------------------------

To add quick filters to a list view, include the ``list_toolbar.html.twig`` template and pass the ``quickFilters`` option:

.. code-block:: twig

   {{ include('@MauticCore/Helper/list_toolbar.html.twig', {
       'searchValue': searchValue,
       'searchHelp': 'mautic.form.form.help.searchcommands',
       'searchId': 'list-search',
       'action': currentRoute,
       'quickFilters': [
           {
               'search': 'has:results',
               'label': 'mautic.core.search.quickfilter.form_results',
               'tooltip': 'mautic.core.search.quickfilter.form_results.tooltip',
               'icon': 'ri-file-list-2-line'
           }
       ]
   }) }}

Quick filter options
--------------------

> Each quick filter is defined as an associative array with the following keys:

Reviewer: ```suggestion
You define each quick filter as an associative array with the following keys:
```

> - ``search``: The search query to be applied

Reviewer: ```suggestion
- ``search``: the search query to apply
```

> - ``label``: The displayed text for the filter button

Reviewer: ```suggestion
- ``label``: the displayed text for the filter button
```

> - ``tooltip``: The tooltip text shown on hover

Reviewer: ```suggestion
- ``tooltip``: the tooltip text shown on hover
```

> - ``icon``: The CSS class for the icon displayed on the button

Reviewer: ```suggestion
- ``icon``: the CSS class for the icon displayed on the button
```
